keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -1,959 +1,1429 @@
+- Support **workload portability** across clusters
+- Achieve all goals in a **vendor neutral** manner
+- Standardize the mechanism for third-party vendors to integrate easily with COSI
+- Allow users (non-admins) to create and utilize buckets (self-service)
+- Establish best-in-class **Access Control** practices for bucket creation and sharing
 
 ### Non-Goals
 
-+ **Data Plane** API standardization will not be addressed by this project
-+ **Bucket Mutation** will not be supported as of now
+- **Data Plane** API standardization will not be addressed by this project
+- **Bucket Mutation** is not supported as of v1alpha2
 
-## COSI architecture
+## Proposal
 
-Since this is an entirely new feature, it is possible to implement this completely out of tree. The following components are proposed for this architecture:
+### User Personas
 
- - COSI ControllerManager
- - COSI Sidecar
- - COSI Driver
+We define 3 kinds of stakeholders:
 
-1. The COSI ControllerManager is the central controller that validates, authorizes and binds COSI created buckets to BucketClaims. Only one active instance of ControllerManager should be present.
-2. The COSI Sidecar is the point of integration between COSI and drivers. All operations that require communication with the OSP is triggered by the Sidecar using gRPC calls to the driver. One active instance of Sidecar should be present **for each driver**.
-3. The COSI driver communicates with the OSP to conduct Bucket related operations.
+- **Administrators** (a.k.a., admin)
+  - Administrators of the Kubernetes cluster's COSI functionality
+  - Deploy, configure, and manage COSI vendor drivers in the Kubernetes cluster
+  - Establish cluster-wide policies for access control over COSI resources
 
-More information about COSI driver is [here](#cosi-driver)
+- **Application developers** (a.k.a., DevOps, user)
+  - End-users of COSI resources claim and use object storage buckets for their applications
+  - Request a bucket to be created and/or provisioned for workloads
+  - Request access to be created/deleted for a specific bucket
+  - Request deletion of a created bucket
 
-## COSI API
+- **Object Storage Providers** (a.k.a., OSP, vendor)
+  - Vendors who offer object storage capabilities for any given object storage system
+  - Comparable to a storage vendor in the CSI domain
+  - Create a COSI driver for their object storage system following the COSI API specs
 
-COSI defines 5 new API types
+## Design details
 
- - [Bucket](#bucket)
- - [BucketClaim](#bucketclaim)
- - [BucketAccess](#bucketaccess)
- - [BucketClass](#bucketclass)
- - [BucketAccessClass](#bucketaccessclass)
+### Important changes between versions
 
-Detailed information about these API types are provided inline with user stories. 
+#### v1alpha1 to v1alpha2
 
-Here is a TL;DR version:
+- DeletionPolicy is now a required field on all API objects. The user must explicitly specify it, leaving no room for confusion.
```

Comment:
How do you convert objects from v1alpha1 to v1alpha2 with a newly required field when the old objects may not have a valid choice?

Reply:
The short is that we don't. v1alpha2 doesn't maintain a goal of supporting v1alpha1 resources